File: backend-v5/core/Enrollment.php
<?php

// Modul: core
// Klasa: Enrollment
// Opis: podaci o statusu studenta na nekom studiju


require_once(Config::$backend_path."core/EnrollmentType.php");
require_once(Config::$backend_path."core/Programme.php");

class Enrollment {
	public $student, $Programme, $semester, $AcademicYear, $EnrollmentType /* provjeriti terminologiju */, $repeat, $Curriculum, $Decision;
	
	// The programme that given student is enrolled in in current academic year
	// If both even and odd semester exist for current year, even (bigger) semester will be returned
	// If student is not enrolled in current year, returns false
	public static function getCurrentForStudent($studentId) {
		$enr = DB::query_assoc("SELECT ss.student student, ss.studij Programme, ss.semestar semester, ss.akademska_godina AcademicYear, ss.nacin_studiranja EnrollmentType, ss.ponovac _repeat, ss.plan_studija Curriculum, ss.odluka Decision FROM student_studij as ss, akademska_godina as ag WHERE ss.student=$studentId AND ss.akademska_godina=ag.id AND ag.aktuelna=1 ORDER BY semestar DESC LIMIT 1");
		if (!$enr) return false;
		
		$enr['repeat'] = $enr['repeat']; unset($enr['repeat']); // reserved word in SQL
		$enr = Util::array_to_class($enr, "Enrollment", array("Programme", "AcademicYear", "EnrollmentType", "Curriculum", "Decision"));
		$enr->student = new UnresolvedClass("Person", $studentId, $enr->student);
		if ($enr->repeat == 1) $enr->repeat=true; else $enr->repeat=false; // FIXME use boolean in database
		return $enr;
	}
	
	// History of all enrollments of given student, sorted by academic year and semester
	public static function getAllForStudent($studentId) {
		$enrolls = DB::query_table("select student, studij Programme, semestar semester, akademska_godina AcademicYear, nacin_studiranja EnrollmentType, ponovac _repeat, plan_studija Curriculum, odluka Decision from student_studij where student=$studentId order by akademska_godina, semestar");

		foreach($enrolls as &$enr) {
			$enr['repeat'] = $enr['repeat']; unset($enr['repeat']); // reserved word in SQL
			$enr = Util::array_to_class($enr, "Enrollment", array("Programme", "AcademicYear", "EnrollmentType", "Curriculum", "Decision"));
			$enr->student = new UnresolvedClass("Person", $studentId, $enr->student);
			if ($enr->repeat == 1) $enr->repeat=true; else $enr->repeat=false; // FIXME use boolean in database
			return $enr;
		}
		return $enrolls;
	}
}

File: rorbackend/app/models/core/enrollment.rb
class Core::Enrollment < ActiveRecord::Base
  # Uncomment following lines if working with legacy database
  # set_table_name 'student_studij'
  # set_primary_key :student, :studij, :semestar, :akademska godina
  # alias_attribute :student_id, :student
  # alias_attribute :programme_id, :studij
  # alias_attribute :semester, :semestar
  # alias_attribute :academic_year_id, :akademska_ godina
  # alias_attribute :enrollment_type_id, :nacin_studiranja
  # alias_attribute :repeat, :ponovac
  # alias_attribute :curriculm_id, :plan_studija
  # alias_attribute :document_id, :odluka

  # Uncomment following lines if working with legacy database
  # TABLE_NAME = 'student_studij'
  # STUDENT_ID = TABLE_NAME + '.' + 'student'
  # PROGRAMME_ID = TABLE_NAME + '.' + 'studij'
  # SEMESTER = TABLE_NAME + '.' + 'semestar'
  # ACADEMIC_YEAR_ID = TABLE_NAME + '.' + 'akademska_godina'
  # ENROLLMENT_TYPE_ID = TABLE_NAME + '.' + 'nacin_studiranja'
  # REPEAT = TABLE_NAME + '.' + 'ponovac'
  # CURRICULM_ID = 'plan_studija'
  # DOCUMENT_ID = 'odluka'

  # Comment following lines if working with legacy database
  TABLE_NAME = 'core_enrollments'
  STUDENT_ID = TABLE_NAME + '.' + 'student_id'
  PROGRAMME_ID = TABLE_NAME + '.' + 'programme_id'
  SEMESTER = TABLE_NAME + '.' + 'semester'
  ACADEMIC_YEAR_ID = TABLE_NAME + '.' + 'academic_year_id'
  ENROLLMENT_TYPE_ID = TABLE_NAME + '.' + 'enrollment_type_id'
  REPEAT = TABLE_NAME + '.' + 'repeat'
  CURRICULM_ID = TABLE_NAME + '.' + 'curriculm_id'
  DOCUMENT_ID = TABLE_NAME + '.' + 'document_id'

  ALL_COLUMNS = [STUDENT_ID, PROGRAMME_ID, SEMESTER, ACADEMIC_YEAR_ID, ENROLLMENT_TYPE_ID, REPEAT, CURRICULM_ID, DOCUMENT_ID]
  
  belongs_to :student
  belongs_to :programme
  belongs_to :academic_year
  belongs_to :curriculum
  belongs_to :document
  has_one :enrollment_type
  
  validates_presence_of :student_id, :programme_id, :academic_year_id, :enrollment_type_id
  
  
  def self.get_current_for_student(student_id)
    current_year = (Core::AcademicYear).current_year  # Current academic year
    raise ActiveRecord::RecordNotFound if (current_year == nil)
    enrollment = (Core::Enrollment).where(:student_id => student_id, :academic_year_id => current_year.id).order(:semester + " DESC").first
      
      return enrollment
  end
  
  
  def self.get_all_for_student(student_id)
    enrollments = (Core::Enrollment).where(:student_id => student_id).order(:academic_year_id, :semester)
    
    return enrollments
  end
end
